File: authentication-library/src/main/java/com/michibaum/authentication_library/JWSValidator.java
package com.michibaum.authentication_library;

import com.auth0.jwt.JWT;
import com.auth0.jwt.JWTVerifier;
import com.auth0.jwt.algorithms.Algorithm;
import com.auth0.jwt.exceptions.JWTVerificationException;

import java.security.interfaces.RSAPublicKey;

public class JWSValidator {

    public Boolean validate(String token, RSAPublicKey publicKey){
        try {
            Algorithm algorithm = Algorithm.RSA256(publicKey, null);
            JWTVerifier verifier = JWT.require(algorithm)
                    .withIssuer("authentication-service")
                    .build(); //Reusable verifier instance
            verifier.verify(token);
            return true;
        } catch (JWTVerificationException exception){
        }
        return false;
    }

}

File: authentication-library/src/main/kotlin/com/michibaum/authentication_library/JWSValidator.kt
package com.michibaum.authentication_library

import com.auth0.jwt.JWT
import com.auth0.jwt.JWTVerifier
import com.auth0.jwt.algorithms.Algorithm
import com.auth0.jwt.exceptions.JWTVerificationException
import java.security.interfaces.RSAPublicKey

open class JWSValidator {
    fun validate(token: String, publicKey: RSAPublicKey): Boolean {
        try {
            val algorithm = JWSAlgorithm.algorithm(publicKey)
            val verifier: JWTVerifier = JWT.require(algorithm)
                    .withIssuer("authentication-service")
                    .build() //Reusable verifier instance
            verifier.verify(token)
            return true
        } catch (_: JWTVerificationException) {
        }
        return false
    }
}